packages/react-components/src/theming/generateTheme.ts

```diff
@@ -74,8 +75,12 @@ const getForegroundColors = (isDark: boolean): Record<string, string> => {
     'black'
   ];
   const baseTheme = isDark ? darkTheme : lightTheme;
-  return foregroundColorKeys.reduce((acc, key) => {
-    acc[key] = baseTheme.palette?.[key];
+  return foregroundColorKeys.reduce((acc: Record<string, string>, key) => {
+    const k = key as keyof IPalette;
```

Comment:
nitpick: Not the biggest deal, I feel like `k` can have more context like `colorCode` or something along those lines. Since that seems to be what keys in IPalette are.